File: automation-service/src/main/java/com/goracz/automationservice/entity/AutomationAction.java
package com.goracz.automationservice.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.ZonedDateTime;

@Document
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class AutomationAction implements Serializable {
    @Id
    private String id;
    private int priority;
    private AutomationActionType type;
    private String appId;
    private String channelId;
    private int volume;
    @CreatedDate
    private ZonedDateTime createdAt;
    @LastModifiedDate
    private ZonedDateTime updatedAt;
}

File: frontend-next/src/models/tv/automation/automation-action.ts
import { AutomationActionType } from './automation-action-type';

export interface AutomationAction {
  id?: string;
  priority: number;
  type: AutomationActionType;
  appId: string;
  channelId: string;
  volume: number;
  createdAt?: Date;
  updatedAt?: Date;
}
